### PR TITLE
feat(blockifier): define alias compressor struct

### PR DESCRIPTION
**Pull request by yoavGrs**

### CODE REVIEW

**Comment by yoavGrs** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2765?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2765** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2765?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2765?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2674** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2674?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2675](https://github.com/starkware-libs/sequencer/pull/2675) <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2675?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2535** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2535?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2534** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/2534?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by dorimedini-starkware — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765#-:-OEPL8CL5SIyr77Ne75M:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: all files reviewed, 2 unresolved discussions (waiting on @nimrod-starkware and @yoavGrs)

**Review by dorimedini-starkware — APPROVED** [review]
> Reviewed 1 of 1 files at r6, all commit messages.
> Dismissed @graphite-app[bot] from 2 discussions.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

File: crates/blockifier/src/state/stateful_compression.rs [hunk 1]
```diff
@@ -14,6 +17,17 @@ pub mod stateful_compression_test;
 type Alias = Felt;
 type AliasKey = StorageKey;
 
+#[derive(Debug, Error)]
+pub enum CompressionError {
+    #[error("Missing key in alias contract: {:#064x}", ***.0)]
```
[the remark this answers is not in the document]
Comment (yoavGrs) [reply crates/blockifier/src/state/stateful_compression.rs:22]: :-1:

File: crates/blockifier/src/state/stateful_compression.rs [hunk 2]
```diff
@@ -123,3 +137,43 @@ impl<'a, S: StateReader> AliasUpdater<'a, S> {
...
+            Ok(self.get_alias(StorageKey(contract_address.0))?.try_into()?)
+        } else {
+            Ok(*contract_address)
+        }
+    }
+
+    fn compress_storage_key(
+        &self,
+        storage_key: &StorageKey,
+        contact_address: &ContractAddress,
```
[the remark this answers is not in the document]
Comment (yoavGrs) [reply crates/blockifier/src/state/stateful_compression.rs]: :+1:

**Review by yoavGrs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

**Review by dorimedini-starkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2765)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @nimrod-starkware)

**Pull request merged**